lib/src/controls/navigation/navigation_view/pane_items.dart

```diff
@@ -679,20 +685,73 @@ class __PaneItemExpanderState extends State<_PaneItemExpander>
     return body.pane?.effectiveIndexOf(widget.item) ?? 0;
   }
 
-  void toggleOpen() {
+  void toggleOpen({bool doFlyout = true}) {
     setState(() => _open = !_open);
 
-    PageStorage.of(context)?.writeState(
+    PageStorage.of(context).writeState(
       context,
       _open,
       identifier: 'paneItemExpanderOpen$index',
     );
-    if (useFlyout) {
-      flyoutController.toggle();
-    }
     if (_open) {
+      if (useFlyout && doFlyout) {
+        final body = InheritedNavigationView.of(context);
+        final displayMode = body.displayMode;
+        final navigationTheme = NavigationPaneTheme.of(context);
+
+        flyoutController.showFlyout(
+          placementMode: displayMode == PaneDisplayMode.compact
+              ? FlyoutPlacementMode.right
+              : FlyoutPlacementMode.bottomCenter,
+          builder: (context) {
+            return MenuFlyout(
+              items: widget.items.map<MenuFlyoutItemBase>((item) {
+                if (item is PaneItem) {
+                  return _PaneItemExpanderMenuItem(
```

Comment:
Consider adding a method like `getMenuItem` to PaneItem so you can remove this whole switch/case statement